### PR TITLE
Add a fipsified base container

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

File: src/bci_build/package.py [hunk 1]
```diff
@@ -2016,6 +2048,50 @@ def _get_os_container_package_names(os_version: OsVersion) -> tuple[str, ...]:
...
+        },
+        custom_end="".join(
+            f"#!RemoteAssetUrl: {_FIPS_15_SP2_ASSET_BASEURL}{binary}\nCOPY {os.path.basename(binary)} .\n"
+            for binary in _FIPS_15_SP2_BINARIES
+        ).strip()
+        + textwrap.dedent(
+            f"""
+            {DOCKERFILE_RUN} \
+                [ $(LC_ALL=C rpm --checksig -v *rpm | \
+                    grep -c  "Header.*key ID 39db7c82: OK") = {len(_FIPS_15_SP2_BINARIES)} ] \
```
Comment (Vogtinator) [inline src/bci_build/package.py:2084]:
```suggestion
                    grep -c "Header.*key ID 39db7c82: OK") = {len(_FIPS_15_SP2_BINARIES)} ] \
```

File: src/bci_build/package.py [hunk 2]
```diff
@@ -2016,6 +2048,50 @@ def _get_os_container_package_names(os_version: OsVersion) -> tuple[str, ...]:
...
+                && rpm -Uvh --oldpackage *openssl*.rpm libgcrypt*rpm \
```
Comment (Vogtinator) [inline src/bci_build/package.py:2085]: You already use `*.rpm` everywhere else

```suggestion
                && rpm -Uvh --oldpackage *.rpm \
```

File: src/bci_build/package.py [hunk 3]
```diff
@@ -2016,6 +2048,50 @@ def _get_os_container_package_names(os_version: OsVersion) -> tuple[str, ...]:
...
+] + [
+    f"SUSE:SLE-15-SP1:Update/pool/x86_64/libgcrypt.15117/{name}-1.8.2-8.36.1.x86_64.rpm"
+    for name in ("libgcrypt20", "libgcrypt20-hmac")
+]
+
+FIPS_BASE_CONTAINERS = [
+    OsContainer(
+        name="base-fips",
+        package_name="base-fips-image",
+        exclusive_arch=[Arch.X86_64],
```
Comment (Vogtinator) [inline src/bci_build/package.py:2066]: Why x86_64 only?

File: src/bci_build/package.py [hunk 4]
```diff
@@ -2016,6 +2048,50 @@ def _get_os_container_package_names(os_version: OsVersion) -> tuple[str, ...]:
...
+        os_version=os_version,
+        build_recipe_type=BuildType.DOCKER,
+        support_level=SupportLevel.L3,
+        is_latest=os_version in CAN_BE_LATEST_OS_VERSION,
+        pretty_name=f"{os_version.pretty_os_version_no_dash} FIPS-140-2",
+        package_list=["fipscheck"],
```
Comment (Vogtinator) [inline src/bci_build/package.py:2072]: Is this not installed already?

File: src/bci_build/package.py [hunk 5]
```diff
@@ -2016,6 +2048,50 @@ def _get_os_container_package_names(os_version: OsVersion) -> tuple[str, ...]:
...
+            "usage": "This container should only be used on a FIPS enabled host (fips=1 on kernel cmdline)."
+        },
+        custom_end="".join(
+            f"#!RemoteAssetUrl: {_FIPS_15_SP2_ASSET_BASEURL}{binary}\nCOPY {os.path.basename(binary)} .\n"
+            for binary in _FIPS_15_SP2_BINARIES
+        ).strip()
+        + textwrap.dedent(
+            f"""
+            {DOCKERFILE_RUN} \
+                [ $(LC_ALL=C rpm --checksig -v *rpm | \
```
Comment (Vogtinator) [inline src/bci_build/package.py:2083]: I'm not entirely sure: Does this verify against the system keyring? If this is just about the signature check the output could be faked

File: src/bci_build/package.py (hunk 4 above)
Comment (dirkmueller) [reply src/bci_build/package.py:2072]: no its not

File: src/bci_build/package.py [hunk 6]
```diff
@@ -2016,6 +2048,50 @@ def _get_os_container_package_names(os_version: OsVersion) -> tuple[str, ...]:
     for os_version in ALL_BASE_OS_VERSIONS
 ]
 
+# https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp3991.pdf
+# Chapter 9.1 Crypto Officer Guidance
+_FIPS_15_SP2_ASSET_BASEURL = "https://api.opensuse.org/public/build/"
```
Comment (Vogtinator) [inline src/bci_build/package.py:2053]: Hm, downloading binaries from build.o.o that way isn't great...

Using an aggregate to let OBS handle this isn't great either though.

File: src/bci_build/package.py (hunk 5 above)
Comment (dirkmueller) [reply src/bci_build/package.py:2083]: it does check against the system keyring and supports the matching keyid (which it greps for)

File: src/bci_build/package.py (hunk 6 above)
Comment (dirkmueller) [reply src/bci_build/package.py:2053]: it works great.

File: src/bci_build/package.py (hunk 3 above)
Comment (dirkmueller) [reply src/bci_build/package.py:2066]: because the rest of the packages are x86_64 specific

File: src/bci_build/package.py (hunk 5 above)
Comment (Vogtinator) [reply src/bci_build/package.py:2083]: Apparently just matching the `HEADER` is not enough: https://bugzilla.suse.com/show_bug.cgi?id=1184501#c16

File: src/bci_build/package.py (hunk 3 above)
Comment (Vogtinator) [reply src/bci_build/package.py:2066]: > the original request is rancher which only exists for that architecture.

ok

File: src/bci_build/package.py (hunk 5 above)
Comment (dirkmueller) [reply src/bci_build/package.py:2083]: @Vogtinator changed it to require the V3 header+payload signature

**Comment by Vogtinator** [comment]
Did the CI set up a test build?

**Comment by dirkmueller** [comment]
> Did the CI set up a test build?

no, the CI is not setup for SP3 anymore. I have testbuild it in IBS and OBS and it works.

**Comment by Vogtinator** [comment]
> > Did the CI set up a test build?
> 
> no, the CI is not setup for SP3 anymore. I have testbuild it in IBS and OBS and it works.

Link?

**Comment by dirkmueller** [comment]
shared on private channel

**Comment by Vogtinator** [comment]
> shared on private channel

Apparently the `textwrap.dedent` didn't quite work (no actual newlines in the code?), other than that it looks ok.

**Pull request merged**